src/EntityFramework/Utilities/DbProviderFactoryExtensions.cs

```diff
@@ -29,6 +30,11 @@ public static string GetProviderInvariantName(this DbProviderFactory factory)
 
             if (row == null)
             {
+                if (factory is SqlClientFactory)
+                {
+                    return "System.Data.SqlClient";
```

Comment:
Ditto for mapping the factory to its invariant name.

Reply:
Same as above.